pkg/utils/gardener/dns.go

```diff
@@ -103,3 +115,19 @@ func GenerateDNSProviderName(secretName, providerType string) string {
 		return ""
 	}
 }
+
+func getIPStackForFamilies(ipFamilies []gardencorev1beta1.IPFamily) string {
+	if gardencorev1beta1.IsIPv4SingleStack(ipFamilies) {
+		return AnnotationValueIPStackIPv4
+	}
+	if gardencorev1beta1.IsIPv6SingleStack(ipFamilies) {
+		return AnnotationValueIPStackIPv6
+	}
+	if len(ipFamilies) == 2 &&
+		((ipFamilies[0] == gardencorev1beta1.IPFamilyIPv4 && ipFamilies[1] == gardencorev1beta1.IPFamilyIPv6) ||
+			(ipFamilies[0] == gardencorev1beta1.IPFamilyIPv6 && ipFamilies[1] == gardencorev1beta1.IPFamilyIPv4)) {
+		return AnnotationValueIPStackIPDualStack
```

Comment:
```suggestion
	if len(ipFamilies) == 2 && slices.Contains(ipFamilies, gardencorev1beta1.IPFamilyIPv4) && slices.Contains(ipFamilies, gardencorev1beta1.IPFamilyIPv6) {
		return AnnotationValueIPStackIPDualStack
```

Reply:
Thanks a lot. That looks a lot nicer and does the same thing.